src/relations/database_provides.py

```diff
@@ -166,6 +166,22 @@ def __init__(self, charm_: "abstract_charm.MySQLRouterCharm") -> None:
         charm_.framework.observe(self._interface.on.database_requested, charm_.reconcile)
         charm_.framework.observe(charm_.on[self._NAME].relation_broken, charm_.reconcile)
 
+    def is_exposed(self, event=None) -> bool:
+        """Whether the relation is exposed."""
+        if event and isinstance(event, ops.RelationEvent):
```

Comment:
if you are testing for the bellow (i.e. testing for all relations), why test for the event relation before?

Reply:
@paulomach there are two questions here:
1) If I am running an event related to a given client, should I get an exposed (i.e. node-port) or not? That is from where `self.is_exposed(relation=...)` comes into play
2) Do I need to expose the mysql-router, at all? This is where `self.is_exposed()` comes into play